common/realtime.py

```diff
@@ -94,3 +94,91 @@ def monitor_time(self) -> bool:
     self._frame += 1
     self._remaining = remaining
     return lagged
+
+class DurationTimer:
```

Comment:
Are these really nessesary?

Reply:
No, but I  think it looks cleaner to use this class for the control frame counter. You can call the `ControlsTimer.tick()` in controlsd instead of doing `self.frame += 1` in every brand's carcontroller update(). Then you can create an arbitrary amount of ControlsTimer objects to have independent duration timers. For example, in mazda/carcontroller.py I need to keep the "HOLD" signal high for 6 seconds so I just create a ControlsTimer object with a duration of 6 seconds and check if it's active(). Then I call the objects reset() to zero the timer. Another use is to call the ControlsTimer.interval(x) which replaces the ugly looking `if self.frame % x == 0`.

So no, it's not necessary but I think it's a nicer interface to use.

Reply:
It might be nice to clean up the car controllers in this way, but let's make that a separate PR

Reply:
Roger that